src/main/scala/scorex/util/serialization/VLQReader.scala

```diff
@@ -0,0 +1,132 @@
+    */
+  @inline override def getUInt(): Long = {
+    val x = getULong()
+    assert(x >= 0L && x <= 0xFFFFFFFFL, s"$x is out of unsigned int range")
+    x
+  }
+
+  /**
+    * Decode signed Long previously encoded with [[VLQByteBufferWriter.putLong]] using VLQ with ZigZag.
+    *
+    * @note Uses ZigZag encoding. Should be used to decode '''only''' a value that was previously
+    *       encoded with [[VLQByteBufferWriter.putLong]].
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @return signed Long
+    */
+  @inline override def getLong(): Long = decodeZigZagLong(getULong())
+
+  /**
+    * Decode Long previously encoded with [[VLQByteBufferWriter.putULong]] using VLQ.
+    * @see [[https://en.wikipedia.org/wiki/Variable-length_quantity]]
+    * @return Long
+    */
+  @inline override def getULong(): Long = {
+    // should be fast if java -> scala conversion did not botched it
+    // source: http://github.com/google/protobuf/blob/a7252bf42df8f0841cf3a0c85fdbf1a5172adecb/java/core/src/main/java/com/google/protobuf/CodedInputStream.java#L2653
+    // for faster version see: http://github.com/google/protobuf/blob/a7252bf42df8f0841cf3a0c85fdbf1a5172adecb/java/core/src/main/java/com/google/protobuf/CodedInputStream.java#L1085
+    var result: Long = 0
+    var shift = 0
+    while (shift < 64) {
+      val b = getByte()
+      result = result | ((b & 0x7F).toLong << shift)
+      if ((b & 0x80) == 0) return result
+      shift += 7
+    }
+    sys.error(s"Cannot deserialize Long value. Unexpected reader $this with bytes remaining ${remaining}")
+    // see https://rosettacode.org/wiki/Variable-length_quantity for implementations in other languages
+  }
+
+  @inline override def getBits(size: Int): Array[Boolean] = {
+    if (size == 0) return Array[Boolean]()
+    val bitSet = BitSet.valueOf(getBytes((size + 7) / 8))
+    val boolArray = new Array[Boolean](size)
+    var i = 0
+    while (i < size) {
+      boolArray(i) = bitSet.get(i)
+      i += 1
+    }
+    boolArray
+  }
+
+  @inline override def getOption[T](getValue: => T): Option[T] = {
+    val tag = getByte()
+    if (tag != 0)
+      Some(getValue)
+    else
+      None
+  }
+
+  /**
+    * Decode String is shorter than 256 bytes
+    *
+    * @return
+    */
+  @inline override def getShortString(): String = {
+    val size = getUByte()
+    new String(getBytes(size))
+  }
+}
+
+object VLQReader {
+
+  /**
+    * Decode a signed value previously ZigZag-encoded with [[VLQByteBufferWriter.encodeZigZagInt]]
+    *
+    * @see [[https://developers.google.com/protocol-buffers/docs/encoding#types]]
+    * @param n unsigned Int previously encoded with [[VLQByteBufferWriter.encodeZigZagInt]]
+    * @return signed Int
+    */
+  def decodeZigZagInt(n: Int): Int =
+  // source: http://github.com/google/protobuf/blob/a7252bf42df8f0841cf3a0c85fdbf1a5172adecb/java/core/src/main/java/com/google/protobuf/CodedInputStream.java#L553
+    (n >>> 1) ^ -(n & 1)
```

Comment:
Why separate function declaration apart from its body with comment line when no braces used `{}`? 
Here and in many places below.

Reply:
fixed